vpplink/binapi/vpp_clone_current.sh

```diff
@@ -1,11 +1,34 @@
 #!/bin/bash
 set -e
 
+SCRIPTDIR="$( cd "$( dirname "${BASH_SOURCE[0]}" )" >/dev/null 2>&1 && pwd )"
+CACHE_DIR=$SCRIPTDIR/.cherries-cache
+
+function green ()
+{
+  printf "\e[0;32m$1\e[0m\n"
+}
+
+function commit_exists ()
+{
+  git show $1 > /dev/null 2>&1 || echo "no"
+}
+
 function git_cherry_pick ()
 {
 	refs=$1
-	git fetch "https://gerrit.fd.io/r/vpp" ${refs}
-	git cherry-pick FETCH_HEAD
+	mkdir -p $CACHE_DIR
+	CACHE_F=$CACHE_DIR/$(echo $refs |sed s@refs/changes/@@g |sed s@/@_@g)
+	if [ -f $CACHE_F ] && [ x$(commit_exists $(cat $CACHE_F)) != xnotfound ]; then
+		COMMIT_HASH=$(cat $CACHE_F)
+		green "Using cached $COMMIT_HASH"
+	else
+		green "Fetching $refs"
+		git fetch "https://gerrit.fd.io/r/vpp" ${refs}
+		COMMIT_HASH=$(git log FETCH_HEAD -1 --pretty=%H)
+	fi
+	git cherry-pick $COMMIT_HASH
+	echo $COMMIT_HASH > $CACHE_F
 	git commit --amend -m "gerrit:${refs#refs/changes/*/} $(git log -1 --pretty=%B)"
```

Comment:
Do we still need the commit after the cherry-pick?
Also did you make sure that this crashed explicitly if the cherry-pick fails to apply cleanly?

Reply:
Right, I added a logic to print some help and cleanly exit after an error in cherry-pick / reverts

Reply:
The `git commit` I use to change the commit message so that the generate.log 
message contain a reference to the gerrit patches.
It's debatable, but I'd argue it's nice to have it here

Reply:
Sounds good 👍